x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -217,6 +249,29 @@ export class ExecuteReportTask implements ReportingTask {
     return await store.setReportFailed(report, doc);
   }
 
+  private async _saveExecutionError(
+    report: SavedReport,
+    failedToExecuteErr: any
+  ): Promise<UpdateResponse<ReportDocument>> {
+    const message = `Saving execution error for ${report.jobtype} job ${report._id}`;
+    const errorParsed = parseError(failedToExecuteErr);
+    // log the error
+    errorLogger(this.logger, message, failedToExecuteErr);
+
+    // update the report in the store
+    const store = await this.getStore();
+    const completedTime = moment().toISOString();
+    const doc: ReportFailedFields = {
+      completed_at: completedTime,
+      output: null,
+      error: errorParsed,
+    };
+
+    // We don't want to mark the report status as failed just yet... we just want to save the error
+    const markReportAsFailed = false;
+    return await store.setReportFailed(report, doc, markReportAsFailed);
```

Comment:
Rather than using this method with an added boolean, I'd prefer if we add a more suitable method on the store that will make more sense to the reader and not need the clarifying comment:
```typescript
    // update the report in the store
    const store = await this.getStore();
    const doc = { output: null, error: errorParsed };
    return await store.setReportError(report, doc);
```

Reply:
Done in https://github.com/elastic/kibana/pull/173775/commits/1b1993d659c0a3daa266e2cf8977fbe023cfd6bc